pyproject.toml

```diff
@@ -245,6 +245,10 @@ filterwarnings = [
     # random resource warnings; I suspect these are coming from vendor SDKs when running examples..
     "ignore:unclosed <socket:ResourceWarning",
     "ignore:unclosed event loop:ResourceWarning",
+    # google-genai SDK creates SSL connections that aren't always properly closed during garbage
+    # collection, triggering PytestUnraisableExceptionWarning in older Python/package versions
+    "ignore:unclosed <ssl.SSLSocket:ResourceWarning",
+    "ignore:.*unclosed.*ssl.SSLSocket.*:pytest.PytestUnraisableExceptionWarning",
```

Comment:
there's a currently a bug either in CI or in the test suite that's causing this, but it's transient and not individual to this PR so please remove this

to trigger CI runs after flaky failures you can push empty commits or let me know so I can restart it

Reply:
no problem, done. btw this pr experienced minor scope creep - there's an existing bug in the test embedding model implementation that was causing the configurable dimension parameter not to be used (so docs using that are also erroneous).

Reply:
fixing tests is always welcome :)